### PR TITLE
allow hostnames in join

### LINKED ISSUE #248: Allowing `addr` to be hostname
What would be the blockers (if any) that need to be resolved to allow `addr` to be a hostname (in `swarm join`)? 

When using SSL, forcing the advertised address (which I understand `swarm manage` will collect to) to be a hostname makes things a little bit more complicated than they should (i.e. you need a certificate with an IP SAN, and generating one with OpenSSL isn't exactly the most intuitive thing either).

Or maybe i'm missing something entirely?!

Happy to help if possible,

Thanks!

### PR DESCRIPTION
**Pull request by vieux**
Closes #248

### CODE REVIEW

File: join.go [hunk 1]
```diff
@@ -21,7 +26,8 @@ func join(c *cli.Context) {
 	}
 
 	addr := c.String("addr")
-	if m, _ := regexp.MatchString("^[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}:[0-9]{1,5}$", addr); !m {
+
+	if !checkAddrFormat(addr) {
 		log.Fatal("--addr should be of the form ip:port")
```
Comment (tnachen) [inline join.go]: Seems like this should be "ip:port or hostname:port"

Comment (vieux) [inline join.go]: absolutly

Comment (vieux) [inline join.go]: updated

**Comment by krallin** [comment]
When I register one my hosts using a hostname instead of an IP, I get the following error in the `swarm manage` logs:

`ERRO[32026] Get my.host.goes.here:2376/v1.15/info: unsupported protocol scheme "my.host.goes.here" `

Passing `tcp://my.host.goes.here:2376` as the `--addr` throws an error, of course. 

It seems `swarm manage` might be doing its own parsing somewhere and isn't able to accept a hostname?

**Comment by aluzzardi** [comment]
LGTM

**Comment by vieux** [comment]
@krallin you need the last version of dockerclient. can you please do a `go get -u ./...` before recompiling

**Comment by krallin** [comment]
Ooh, that makes sense. Thanks, @vieux !

**Pull request merged**